src/execution/compiler/operator/sort_translator.cpp

```diff
@@ -186,11 +194,21 @@ ast::FunctionDecl *SortTranslator::GenerateEndTLSortHookFunction() const {
     auto *sorter_size = codegen->CallBuiltin(ast::Builtin::SorterGetTupleCount, {local_sorter_.GetPtr(codegen)});
     builder.Append(codegen->DeclareVarWithInit(num_tuples, sorter_size));
 
+    brain::ExecutionOperatingUnitType build_ou_type;
+    ast::Expr *cardinality_val;
+    if (const auto &plan = GetPlanAs<planner::OrderByPlanNode>(); plan.HasLimit()) {
+      build_ou_type = brain::ExecutionOperatingUnitType::PARALLEL_SORT_TOPK_MERGE_STEP;
```

Comment:
This should be `brain::ExecutionOperatingUnitType::PARALLEL_SORT_TOPK_STEP` with the else case taking on `brain::ExecutionOperatingUnitType::PARALLEL_SORT_STEP`.

`EndTLSortHook` is the hook called to end a thread local sort.

Reply:
Fixed!